packages/datadog-plugin-rhea/src/index.js

```diff
@@ -0,0 +1,146 @@
+function createWrapSend (tracer, config) {
+  return function wrapSend (send) {
+    return function sendWithTrace (msg, tag, format) {
+      const name = this.options.target && this.options.target.address
+        ? this.options.target.address : 'amq.topic'
+      return tracer.trace('rhea.sender.send', { tags: {
+        'resource.name': name,
+        'service.name': config.service || `${tracer._service}-amqp`,
+        'span.kind': 'producer',
+        'amqp.link.target.address': name,
+        'amqp.link.role': 'sender',
+        'out.host': this.connection.options.host,
+        'out.port': this.connection.options.port
+      } }, (span, done) => {
+        msg.delivery_annotations = msg.delivery_annotations || {}
+        tracer.inject(span, 'text_map', msg.delivery_annotations)
+        const delivery = send.apply(this, arguments)
+        if (this.options.snd_settle_mode !== 1) {
+          delivery._dd = { done, span }
+        } else {
+          done()
+        }
+        return delivery
+      })
+    }
+  }
+}
+
+function createWrapConnectionDispatch (tracer, config) {
+  return function wrapDispatch (dispatch) {
+    return function dispatchWithTrace (eventName, obj) {
+      if (eventName === 'disconnected') {
+        for (const key in this.local_channel_map) {
+          const session = this.local_channel_map[key]
+          const addTags = entry => {
+            if (entry && entry._dd) {
+              const { span, done } = entry._dd
+              const error = obj.error || this.saved_error
+              if (error) {
+                span.addTags({
+                  'error.type': error.name,
+                  'error.msg': error.message,
+                  'error.stack': error.stack
+                })
+              }
+              done()
+            }
+          }
+          session.incoming.deliveries.entries.forEach(addTags)
```

Comment:
There should be safety nets in case these properties don't exist. There should probably also be a check for this when creating the span to make sure it cannot result in a memory leak if the spans can never be finished.